File: LAP4/Lap1/app/src/main/java/com/example/lab1/MainActivity.java
package com.example.lab1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    Button btnCalculator, btnCamera, btnRandomNumber, btnDiceRoller, btnCallSms;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Khởi tạo các button
        btnCalculator = findViewById(R.id.btnCalculator);
        btnCamera = findViewById(R.id.btnCamera);
        btnRandomNumber = findViewById(R.id.btnRandomNumber);
        btnDiceRoller = findViewById(R.id.btnDiceRoller);
        btnCallSms = findViewById(R.id.btnCallSms);

        // Set onClick listeners
        btnCalculator.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, CalculatorActivity.class);
                startActivity(intent);
            }
        });

        btnCamera.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, CameraActivity.class);
                startActivity(intent);
            }
        });

        btnRandomNumber.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, RandomNumberActivity.class);
                startActivity(intent);
            }
        });

        btnDiceRoller.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, DiceRollerActivity.class);
                startActivity(intent);
            }
        });

        btnCallSms.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, CallSmsActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: THINH_LAP5/AccountApp/app/src/main/java/com/example/account/ui/MainActivity.kt
package com.example.account.ui

import android.content.Context
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.account.R
import com.example.account.ui.login.LoginFragment
import com.example.account.ui.register.RegisterFragment

class MainActivity : AppCompatActivity() {

    companion object {
        const val SAVE_PREF = "save_pref"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (savedInstanceState == null) {
            gotoLoginScreen()
        }
    }

    fun gotoRegisterScreen() {
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container_view, RegisterFragment())
            .addToBackStack(null)
            .commit()
    }

    fun gotoLoginScreen() {
        supportFragmentManager.popBackStack()
        supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container_view, LoginFragment())
            .commit()
    }

    fun showToast(message: String) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
    }

    fun saveLastLogin(email: String) {
        val pref = getSharedPreferences(SAVE_PREF, Context.MODE_PRIVATE)
        pref.edit().putString("last_email", email).apply()
    }
}

